src/components/metadataFormControls/ModelSingleSelect/BaseModelSingleSelect.tsx

```diff
@@ -39,7 +39,7 @@ export const BaseModelSingleSelect = <
     const { allModelsMap, allSingleSelectOptions } = useMemo(() => {
         const allModelsMap = new Map(available.map((o) => [o.id, o]))
         // due to pagination, the selected model might not be in the available list, so add it
-        if (selected && !allModelsMap.get(selected.id)) {
+        if (selected && selected.id && !allModelsMap.get(selected.id)) {
```

Comment:
How can the `selected` be truthy but not contain an `id` with the types here? This is more protective and it can stay, but it shouldn't really happen with correct typing.

Reply:
ah i added it cause i messed everything up by sending selected=undefined. But you are right, prob if i fixed teh types before that it would not have happened